src/aggregated_range_proof/dealer.rs

```diff
@@ -13,11 +13,12 @@ pub struct Dealer {}
 
 impl Dealer {
     /// Creates a new dealer coordinating `m` parties proving `n`-bit ranges.
-    pub fn new(
+    pub fn new<'a, 'b>(
+        gens: GeneratorsView<'b>,
         n: usize,
         m: usize,
-        transcript: &mut ProofTranscript,
-    ) -> Result<DealerAwaitingValueCommitments, &'static str> {
+        transcript: &'a mut ProofTranscript,
+    ) -> Result<DealerAwaitingValueCommitments<'a, 'b>, &'static str> {
```

Comment:
Does this mean we cannot serialize the dealer? Seems a bit limiting if the dealer object must remain in memory tied to a lifetime of its parameters.

Reply:
What if you move the transcript in, and then return it at the end, so the outer protocol can continue using it?

Reply:
Yes, this means you cannot serialize the dealer. I think that's fine, since the dealer is created once per run of the (online) aggregation protocol, and I don't think there's a good reason to stop in the middle of the protocol, save the state, and then resume it later -- you could just start over if you needed to do that.

Reply:
makes sense, 👍